### ISSUE TITLE
In some data i get the a different color image tiles apart from true color i  .  

### ISSUE THREAD

**Issue by anup39** [DHI/terracotta#339]
![Screenshot 2024-06-06 at 17 14 35](https://github.com/DHI/terracotta/assets/61059165/1917e6a6-c019-42dd-8794-5e336ae31c1f)

**Comment by dionhaefner** [comment]
Do you see any errors in the server log?

**Comment by anup39** [comment]
In flask application there are no errors.  The endpoint is serving image properly. However i am using gunicorn and nginx to serve the application. Whenever i increase the worker count i am not getting this issue. 
<img width="455" alt="Screenshot 2024-06-07 at 08 46 42" src="https://github.com/DHI/terracotta
<img width="529" alt="Screenshot 2024-06-07 at 08 49 14" src="https://github.com/DHI/terracotta/assets/61059165/b02385a0-7d4e-4b0b-b610-52c9edd5d921">
/assets/61059165/a2817e50-6880-4466-8cb3-c30cddab2361">

**Comment by dionhaefner** [comment]
What's your setup? Assuming the server is running on a single VM? Where are the images (local disk, S3, ...)?

**Comment by anup39** [comment]
I am running in Azure VM with 4 CPU core, RAM 16 GB and with 128 GB SSD for storage , Yes the tifs bands are in my local storage.

**Comment by j08lue** [comment]
Looks like one of the bands is failing or gets an unexpected offset.

> Whenever i increase the worker count i am not getting this issue

That is strange. Is that behaviour consistent? And at which worker size?

And does it change when you [disable multiprocessing](https://terracotta-python.readthedocs.io/en/latest/settings.html#terracotta.config.TerracottaSettings.USE_MULTIPROCESSING) with `USE_MULTIPROCESSING=false`?

Thinking about this part here - reading each band in its own process...

https://github.com/DHI/terracotta/blob/8632e1c598a22de2d10c6cdb885cd4f1b6fe9d8e/terracotta/drivers/geotiff_raster_store.py#L29-L48

If one of the bands fails, the whole tile production should fail, though. And Gunicorn workers should not affect Python processes, should they? 🤔 

Sorry for all the guesswork. Are you sure your file is ok, @anup39? For example, try opening it in QGIS or so and zoom in and out.

**Comment by dionhaefner** [comment]
> If one of the bands fails, the whole tile production should fail, though.

Correct, so this could only happen if the data being read look OK but is faulty (corrupt overview, data being truncated while read), or through a race condition or cache pollution somewhere. Which would be almost impossible to debug without a reproducer...
